controllers/vaultdynamicsecret_controller.go

```diff
@@ -84,14 +84,29 @@ func (r *VaultDynamicSecretReconciler) Reconcile(ctx context.Context, req ctrl.R
 		if apierrors.IsNotFound(err) {
 			return ctrl.Result{}, nil
 		}
-
 		logger.Error(err, "error getting resource from k8s", "obj", o)
 		return ctrl.Result{}, err
 	}
+	// Add a finalizer on the VDS resource if we intend to Revoke on cleanup path, otherwise
+	// there isn't a need for it since we are not managing anything on deletion.
+	if o.Spec.Revoke {
+		if o.GetDeletionTimestamp() == nil {
+			if err := r.addFinalizer(ctx, o); err != nil {
+				return ctrl.Result{}, err
+			}
+		} else {
+			logger.Info("Got deletion timestamp", "obj", o)
+			if err := r.handleDeletion(ctx, o); err != nil {
+				msg := "Failed to handle deletion"
+				r.Recorder.Eventf(o, corev1.EventTypeWarning, consts.ReasonSecretLeaseRevoked, msg+": %s", err)
```

Comment:
I wonder if it would be better to use a reason that's not in the past tense here, since in this err handling the lease may not have been revoked yet. And it looks like this error could be either from removing the finalizer or revoking the lease.

It may be useful to emit an event on successful revoke too.

Reply:
Good point thanks, fixed in 451a3268045f475de39ef228db34c225ff4b5512